src/models/generated/HydroReservoir.jl

```diff
@@ -53,71 +52,68 @@ mutable struct HydroReservoir <: Device
-    HydroReservoir(name, available, storage_level_limits, initial_level, spillage_limits, inflow, outflow, level_targets, travel_time, intake_elevation, LinearCurve(head_to_volume_factor), reservoir_location, operation_cost, level_data_type, ext, InfrastructureSystemsInternal(), )
+function HydroReservoir(name, available, storage_level_limits, initial_level, spillage_limits, inflow, outflow, level_targets, travel_time, intake_elevation, head_to_volume_factor, upstream_turbines=Device[], downstream_turbines=Device[], operation_cost=HydroReservoirCost(nothing), level_data_type=ReservoirDataType.USABLE_VOLUME, ext=Dict{String, Any}(), )
+    HydroReservoir(name, available, storage_level_limits, initial_level, spillage_limits, inflow, outflow, level_targets, travel_time, intake_elevation, head_to_volume_factor, upstream_turbines, downstream_turbines, operation_cost, level_data_type, ext, InfrastructureSystemsInternal(), )
 end
 
-function HydroReservoir(; name, available, storage_level_limits, initial_level, spillage_limits, inflow, outflow, level_targets, travel_time, intake_elevation, head_to_volume_factor, reservoir_location = ReservoirLocation.HEAD, operation_cost = HydroReservoirCost(nothing), level_data_type=ReservoirDataType.USABLE_VOLUME, ext=Dict{String, Any}(), internal=InfrastructureSystemsInternal(), )
-    HydroReservoir(name, available, storage_level_limits, initial_level, spillage_limits, inflow, outflow, level_targets, travel_time, intake_elevation, head_to_volume_factor, reservoir_location, operation_cost, level_data_type, ext, internal, )
+function HydroReservoir(; name, available, storage_level_limits, initial_level, spillage_limits, inflow, outflow, level_targets, travel_time, intake_elevation, head_to_volume_factor, upstream_turbines=Device[], downstream_turbines=Device[], operation_cost=HydroReservoirCost(nothing), level_data_type=ReservoirDataType.USABLE_VOLUME, ext=Dict{String, Any}(), internal=InfrastructureSystemsInternal(), )
+    HydroReservoir(name, available, storage_level_limits, initial_level, spillage_limits, inflow, outflow, level_targets, travel_time, intake_elevation, head_to_volume_factor, upstream_turbines, downstream_turbines, operation_cost, level_data_type, ext, internal, )
 end
 
+# Constructor for demo purposes; non-functional.
 function HydroReservoir(::Nothing)
     HydroReservoir(;
         name="init",
         available=false,
-        initial_level=0.0,
         storage_level_limits=(min=0.0, max=0.0),
+        initial_level=0.0,
         spillage_limits=nothing,
         inflow=0.0,
         outflow=0.0,
-        level_targets=0.0,
-        travel_time=0.0,
+        level_targets=nothing,
+        travel_time=nothing,
         intake_elevation=0.0,
```

Comment:
This is only relevant for the downstream turbine?

Reply:
yeah, travel time needs to be moved to the turbines now